web/src/TracingUI.tsx

```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+import { Switch, Route } from 'react-router-dom';
+import { TraceDetailPage } from './pages/TraceDetailPage';
+import { TracesPage } from './pages/TracesPage';
+
+export default function TracingUI() {
+  return (
+    <Switch>
+      <Route path="/observe/traces" exact component={TracesPage} />
+      <Route
+        path="/observe/traces/:traceId"
+        exact
+        component={TraceDetailPage}
+      />
+      <Route component={TracesPage} />
```

Comment:
Is this needed if we already have route to the TracesPage  <Route path="/observe/traces" exact component={TracesPage} /> ?

Reply:
I kept it here as a catch-all route, if the above routes don't match.
But on second thought, you're right, probably we don't want a catch-all route.
I'll remove it.